test/cmp_client_test.c

```diff
@@ -372,6 +372,39 @@ static int test_try_certreq_poll_abort(void)
     return result;
 }
 
+static int test_exec_GENM_ses_poll(int check_after, int poll_count,
+                                   int total_timeout, int expect)
+{
+    SETUP_TEST_FIXTURE(CMP_SES_TEST_FIXTURE, set_up);
+    fixture->expected = expect;
+    ossl_cmp_mock_srv_set_checkAfterTime(fixture->srv_ctx, check_after);
+    ossl_cmp_mock_srv_set_pollCount(fixture->srv_ctx, poll_count);
+    OSSL_CMP_CTX_set_option(fixture->cmp_ctx,
+                            OSSL_CMP_OPT_TOTAL_TIMEOUT, total_timeout);
+    EXECUTE_TEST(execute_exec_GENM_ses_test, tear_down);
+    return result;
+}
```

Comment:
Please do not copy&paste whole chunks of code - this just leads to redundancy and easily causes inconsistencies, like happended here.

Better generalize  the existing `test_exec_IR_ses_poll()` with a request type parameter and use it for both `ir` and `genm`.